Makefile

```diff
@@ -44,6 +44,7 @@ build_%: ARCHARG = $(if $(ARCH), --arch $(ARCH))
 build_%: GOARG = $(if $(CROSSGOPATH), --go $(CROSSGOPATH))
 build_%: GOBUILD = $(if $(CROSSGOPATH), $(CROSSGOPATH), go)
 build_%: .pre-build
+	$(GOBUILD) clean -cache
```

Comment:
I don't think this is a good idea. The go cache is why builds are fast, and that we need this is a real bug

Reply:
yeah was just discussing more with James, would you be open to something more like: 
`if [[ "${GOCLEANCACHE}" -eq 1 ]]; then $(GOBUILD) clean -cache; fi`
or the simplest for now, just telling people the can run cache clean until we get to the bottom of the linking (I was under the impression this was an ongoing issue because it's happened since I started)

Reply:
i can live with just remembering to run `go clean -cache` until we figure it out

Reply:
that makes sense to me, I will close this. Thanks for the feedback all!